core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/StackV4Controller.java

```diff
@@ -214,9 +217,19 @@ public FlowIdentifier putStartInternal(Long workspaceId, String name, @Initiator
 
     @Override
     @InternalOnly
-    public FlowIdentifier rotateSaltPasswordInternal(Long workspaceId, String crn, @InitiatorUserCrn String initiatorUserCrn) {
-        return stackOperations.rotateSaltPassword(NameOrCrn.ofCrn(crn), restRequestThreadLocalService.getAccountId(),
-                RotateSaltPasswordReason.MANUAL);
+    public FlowIdentifier rotateSaltPasswordInternal(Long workspaceId, String crn, RotateSaltPasswordRequest request,
+            @InitiatorUserCrn String initiatorUserCrn) {
+        RotateSaltPasswordReason rotateSaltPasswordReason = RotateSaltPasswordReason.valueOf(request.getReason().name());
+        return stackOperations.rotateSaltPassword(NameOrCrn.ofCrn(crn), restRequestThreadLocalService.getAccountId(), rotateSaltPasswordReason);
+    }
+
+    @Override
+    @InternalOnly
+    public SaltPasswordStatusResponse getSaltPasswordStatus(Long workspaceId, String crn) {
```

Comment:
`crn` should be annotated with `@TenantAwareParam`, without it, internal call won't work probably